lib/routes/addRelation.js

```diff
@@ -31,7 +31,7 @@ addRelationRoute.register = function() {
         resourceConfig.handlers.find(request, callback);
       },
       function(ourResource, callback) {
-        theirResource = ourResource;
+        theirResource = JSON.parse(JSON.stringify(ourResource));
```

Comment:
This closes a bug identified by the test harness improvements, whereby we were modifying an object provided to us by the MemoryHandler, thus mutating the object in our store prematurely.